pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -171,10 +174,12 @@ func (rc *reconciler) reconcile() {
 	// Ensure volumes that should be unmounted are unmounted.
 	for _, mountedVolume := range rc.actualStateOfWorld.GetMountedVolumes() {
 		if !rc.desiredStateOfWorld.PodExistsInVolume(mountedVolume.PodName, mountedVolume.VolumeName) {
-			// Volume is mounted, unmount it
-			glog.V(12).Infof(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
-			err := rc.operationExecutor.UnmountVolume(
-				mountedVolume.MountedVolume, rc.actualStateOfWorld)
+			volumeHandler, err := operationexecutor.NewVolumeHandler(mountedVolume.VolumeSpec, rc.operationExecutor)
```

Comment:
I don't understand this pattern. Why is a a `NewVolumeHandler` constructed before each usage?

Reply:
`NewVolumeHandler` chooses a handler for filesystem or block volume based on `volumeMode` in mountedVolume.VolumeSpec. Since the volumeMode differs on each volume, we need to choose proper handler for a volume before each unmount volume operation.